src/map/storage.c

```diff
@@ -887,6 +993,145 @@ static int storage_guild_storage_quit(struct map_session_data *sd, int flag)
+ * @retval false in case of error.
+ */
+bool storage_config_read(const char* filename, bool imported)
+{
+	nullpo_retr(false, filename);
+
+	if (!imported)
+		VECTOR_INIT(storage->configuration);
+
+	struct config_t stor_libconf;
+	if (libconfig->load_file(&stor_libconf, filename) == CONFIG_FALSE)
+		return false; // Error message is already shown by libconfig->load_file()
+
+	const struct config_setting_t* setting = NULL;
+	if ((setting = libconfig->setting_get_member(stor_libconf.root, "storage_conf")) == NULL && !imported) {
+		ShowError("storage_config_read: Error in reading file '%s'\n", filename);
+		libconfig->destroy(&stor_libconf);
+		return false;
+	}
+
+	struct config_setting_t* t = NULL;
+	int i = 0;
+	while ((t = libconfig->setting_get_elem(setting, i++)) != NULL) {
+		struct storage_settings s_conf = { 0 };
+		const char *constant = NULL;
+
+		/* Id */
+		if (libconfig->setting_lookup_int(t, "Id", &s_conf.uid) == CONFIG_FALSE) {
+			ShowError("storage_config_read: Id field not found for storage configuration in '%s'. Skipping...\n", filename);
+			continue;
+		}
+
+		// Duplicate ID search and report...
+		int d = 0;
+		ARR_FIND(0, VECTOR_LENGTH(storage->configuration), d, VECTOR_INDEX(storage->configuration, d).uid == s_conf.uid);
+		if (d < VECTOR_LENGTH(storage->configuration)) {
+			ShowError("storage_config_read: Duplicate ID %d for storage. Skipping...\n", s_conf.uid);
+			continue;
+		}
+
+		// Check for an invalid ID...
+		if (s_conf.uid <= 0) {
+			ShowError("storage_config_read: ID for storage cannot be less than or equal to zero. Skipping...\n");
+			continue;
+		}
+
+		/* Name */
+		if (libconfig->setting_lookup_mutable_string(t, "Name", s_conf.name, NAME_LENGTH) == CONFIG_FALSE) {
+			ShowError("storage_config_read: Name field not found for storage configuration (Id: %d) in '%s'. Skipping...\n", s_conf.uid, filename);
+			continue;
+		}
+
+		if (libconfig->setting_lookup_string(t, "Constant", &constant) == CONFIG_FALSE) {
+			ShowError("storage_config_read: Constant field not found for storage configuration (Id: %d) in '%s'. Skipping...\n", s_conf.uid, filename);
+			continue;
+		} else {
+			script->set_constant(constant, s_conf.uid, false, false);
+		}
+
+		/* Capacity */
+		if (libconfig->setting_lookup_int(t, "Capacity", &s_conf.capacity) == CONFIG_FALSE) {
+			ShowError("storage_config_read: Capacity field not found for storage configuration (Id: %d)  in '%s'. Skipping...\n", s_conf.uid, filename);
+			continue;
+		}
+
+		if (s_conf.capacity > MAX_STORAGE) {
+			ShowWarning("storage_config_read: Capacity for Storage #%d ('%s') is over MAX_STORAGE. Capping to %d.\n", s_conf.uid, s_conf.name, MAX_STORAGE);
+			s_conf.capacity = min(s_conf.capacity, MAX_STORAGE);
+		}
+
+		/* Additional Fields */
+		storage->config_read_additional_fields(t, &s_conf, filename);
+
+		if (imported) {
+			ARR_FIND(0, VECTOR_LENGTH(storage->configuration), i, VECTOR_INDEX(storage->configuration, i).uid == s_conf.uid);
```

Comment:
Same as above:
I've been thinking. I wanna drop this feature altogether. Since storage.conf will be as is almost forever officially similar to other configs. There is no point having additional fields and imports.

Reply:
9f08c77a4a4a41c9ea0bd85a9ff678043961edb0

Added check on the first duplicate check.